monty_profiling/floppy/flop_counting/operations/linalg.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from typing import Any
+
+__all__ = ["NormOperation", "CondOperation", "InvOperation", "EigOperation"]
+
+
+class NormOperation:
+    """FLOP count for np.linalg.norm operation."""
+
+    def count_flops(self, *args: Any, result: Any) -> int:
+        """Count FLOPs for norm calculation.
+
+        For Frobenius/L2 norm:
+        - n multiplications for squares
+        - (n-1) additions for sum
+        - 1 square root
+        Total: 2n FLOPs
+        """
+        return 2 * np.size(args[0])
+
+
+class CondOperation:
+    """FLOP count for np.linalg.cond operation."""
+
+    def count_flops(self, *args: Any, result: Any) -> int:
+        """Count FLOPs for condition number calculation.
+
+        Condition number requires:
+        - SVD decomposition (~14n³ for an nxn matrix)
```

Comment:
Did you look up particular references for these elements? If so, it would be helpful to just link to them as a sanity check. Similar for e.g. `LU decomposition (~2/3 n³)`

Reply:
Ok perfect, thanks for the detailed response! Is it possible to then add to the docstring something like

```
An estimate for complexity of SVD is 
~2mn^2  + 11n^3
per equation 11.22 in "Numerical Linear Algebra" by Trefethen and Bau

Here we assume m=n (a square matrix), and so estimate FLOPs as
 14n^3
```

And good to know about efficiency of different operations, but agree it's not a priority at the moment given how often we use SVD.

Reply:
Re. the ord=2, it seems to depend on whether the input is a vector or matrix, see the table here (https://numpy.org/doc/2.1/reference/generated/numpy.linalg.norm.html)

Summarized:
![Uploading Screenshot 2025-01-15 at 12.48.51 PM.png…]()

Frobenius norm is performed when `ord=fro`